### ISSUE TITLE
Bug when using [<ReactComponent>] on a static member function with optional parameters

### ISSUE THREAD

**Issue by Lenne231** [Zaid-Ajaj/Feliz#310]
Hi,

I found a bug when using `[<ReactComponent>]` on a static member function with optional parameters.

Feliz 1.32.0
Fable 3.1.1

### Expected behavior
```fsharp
[<AbstractClass;Sealed>]
type Components =
  [<ReactComponent>]
  static member Header (title: string, ?className: string) =
    Html.h1 [
      if className.IsSome then prop.className className.Value
      prop.children [ Html.text title ]
    ]

let header_not_ok = Components.Header("title")
let header_ok = Components.Header("title", "test")
```
compiles to

```javascript
export class Components {
    constructor() {
    }
}

export function Components$reflection() {
    return class_type("Test.Components", void 0, Components);
}

export function Components_Header_68C4AEB5(components_Header_68C4AEB5InputProps) {
    const className = components_Header_68C4AEB5InputProps.className;
    const title = components_Header_68C4AEB5InputProps.title;
    return createElement("h1", createObj(ofSeq(delay(() => append((className != null) ? singleton(["className", value_4(className)]) : empty(), delay(() => singleton(["children", reactApi.Children.toArray([title])])))))));
}

export const header_not_ok = createElement(Components_Header_68C4AEB5, {
    title: "title",
    className: null,
});

export const header_ok = createElement(Components_Header_68C4AEB5, {
    title: "title",
    className: "test",
});
```

### Actual behavior

It compiles to

```javascript
export class Components {
    constructor() {
    }
}

export function Components$reflection() {
    return class_type("Test.Components", void 0, Components);
}

export function Components_Header_68C4AEB5(components_Header_68C4AEB5InputProps) {
    const className = components_Header_68C4AEB5InputProps.className;
    const title = components_Header_68C4AEB5InputProps.title;
    return createElement("h1", createObj(ofSeq(delay(() => append((className != null) ? singleton(["className", value_4(className)]) : empty(), delay(() => singleton(["children", reactApi.Children.toArray([title])])))))));
}

export const header_not_ok = Components_Header_68C4AEB5("title");

export const header_ok = createElement(Components_Header_68C4AEB5, {
    title: "title",
    className: "test",
});
```

**Comment by Zaid-Ajaj** [comment]
Hi @Lenne231 thanks a lot for the issue report, I think this happens because of [this check here](https://github.com/Zaid-Ajaj/Feliz/blob/master/Feliz.CompilerPlugins/ReactComponent.fs#L23) I will have a look ASAP because this pattern is quite useful

**Comment by Zaid-Ajaj** [comment]
Hi @Lenne231 I just published Feliz v1.34 which should include the fix 🚀 can you please give it a try a let me know if it worked for you? Please reopen the issue if the problem persists :smile:
